pkg/client/client.go

```diff
@@ -1004,9 +1017,35 @@ func monitorKeepTerminatedEvent(
 					members,
 				)
 
-				tss.BroadcastRecoveryAddress(
+				if err != nil {
+					logger.Errorf(
+						"failed to retrieve member ids on keep [%s] termination: [%v]",
+						keep.ID().Hex(),
+						err,
+					)
+					return
+				}
+
+				var beneficiaryAddress string
+				_, err = btcutil.DecodeAddress(bitcoinConfig.BeneficiaryAddress, &chaincfg.TestNet3Params)
+				if err != nil {
+					beneficiaryAddress, err = recovery.DeriveAddress(bitcoinConfig.BeneficiaryAddress, 0)
+				} else {
+					beneficiaryAddress = bitcoinConfig.BeneficiaryAddress
+				}
+				if err != nil {
+					logger.Errorf(
+						"failed to retrieve a btc address from config - keep: [%s] address: [%s] err: [%v]",
+						keep.ID().Hex(),
+						bitcoinConfig.BeneficiaryAddress,
+						err,
+					)
+					return
+				}
```

Comment:
Thoughts on pulling this whole logic to a separate function that would just output the final address? Additionally, we could document the new function to explain what the heck we're doing with the address decode and fallbacks.